sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -408,6 +408,37 @@ object DataType {
     }
   }
 
+  /**
+   * Check if `from` is equal to `to` type except for collations and nullability, which are
+   * both checked to be compatible so that data of type `from` can be interpreted as of type `to`.
+   */
+  private[sql] def equalsIgnoreCompatibleCollationAndNullability(
```

Comment:
let's already give this a name that reflects that is returns true of the it is allowed to evolve the type. This will be true for more cases than collations in the future

Reply:
Please check https://github.com/apache/spark/pull/46734#discussion_r1615598800. Part of the effort was to change the name of this method to `equalsIgnoreCompatibleCollation`.